base/inference.jl

```diff
@@ -279,32 +279,33 @@ const getfield_tfunc = function (A, s0, name)
             if is(snames[i],fld)
                 R = s.types[i]
                 if length(s.parameters) == 0
-                    return R
+                    return R, true
                 else
-                    return limit_type_depth(R, 0, true,
+                    typ = limit_type_depth(R, 0, true,
                                             filter!(x->isa(x,TypeVar), Any[s.parameters...]))
+                    return typ, isleaftype(s) && typeseq(typ, R)
                 end
             end
         end
-        return Bottom
+        return Bottom, true
     elseif isa(A[2],Int)
         if isa(A[1],Module) || s === Module
-            return Bottom
+            return Bottom, true
         end
         i::Int = A[2]
         nf = s.types.length
         if isvatuple(s) && i >= nf
-            return s.types[nf].parameters[1]
+            return s.types[nf].parameters[1], false
         end
         if i < 1 || i > nf
-            return Bottom
+            return Bottom, true
         end
-        return s.types[i]
+        return s.types[i], false
```

Comment:
This can be true, but should probably be merged with Symbol case above in any case.